frame/system/src/migrations/mod.rs

```diff
@@ -0,0 +1,142 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Migrate the reference counting state.
+
+use codec::{Decode, Encode, FullCodec};
+use frame_support::{
+	pallet_prelude::ValueQuery, traits::PalletInfoAccess, weights::Weight, Blake2_128Concat,
+	RuntimeDebug,
+};
+use sp_std::prelude::*;
+
+/// Type used to encode the number of references an account has.
+type RefCount = u32;
+
+/// Information of an account.
+#[derive(Clone, Eq, PartialEq, Default, RuntimeDebug, Encode, Decode)]
+struct AccountInfo<Index, AccountData> {
+	nonce: Index,
+	consumers: RefCount,
+	providers: RefCount,
+	sufficients: RefCount,
+	data: AccountData,
+}
+
+/// Trait to implement to give information about types used for migration
+pub trait V2ToV3 {
+	/// The system pallet.
+	type Pallet: 'static + PalletInfoAccess;
+
+	/// System config account id
+	type AccountId: 'static + FullCodec;
+
+	/// System config index
+	type Index: 'static + FullCodec + Copy;
+
+	/// System config account data
+	type AccountData: 'static + FullCodec;
+}
+
+// ### Warning
+//
+// The call below is only valid because the name System is enforced
+// at runtime construction level for the system pallet.
+frame_support::generate_storage_alias!(
+	System, UpgradedToU32RefCount => Value<
+		bool,
+		ValueQuery
+	>
+);
+
+// ### Warning
+//
+// The call below is only valid because the name System is enforced
+// at runtime construction level for the system pallet.
+frame_support::generate_storage_alias!(
+	System, UpgradedToTripleRefCount => Value<
+		bool,
+		ValueQuery
+	>
+);
+
+// ### Warning
+//
+// The call below is only valid because the name System is enforced
+// at runtime construction level for the system pallet.
+frame_support::generate_storage_alias!(
+	System, Account<T: V2ToV3> => Map<
+		(Blake2_128Concat, T::AccountId),
+		AccountInfo<T::Index, T::AccountData>
+	>
+);
+
+#[allow(dead_code)]
```

Comment:
```suggestion
```

Am I wrong that this is not needed ? because the item is a public item of the crate, so there shouldn't be dead code.

Reply:
Yes, I have removed them all since now the functions are public (I just checked it compiled for good order).